Add pushlog feature for all branches (#51)

Cai0407 opened the pull request:
extended patch for #185

xabolcs commented:
Thanks @Cai0407, will review today or tomorrow!

xabolcs commented:
How about removing `nightly.isTrunk()` part  from the `if` at [line 134][1] instead morphing `nightly.isTrunk()`?


[1]: /mozilla/nightlytt/blob/497220120678f574d317a4ac576c5c7fd4b46bb1/extension/chrome/content/nightly.js#L134

Cai0407 commented:
Hi @xabolcs, thanks for good suggestion. I have updated the patch.
I also removed nightly.variables.platformversion.indexOf("pre") from line 67 to be available for minor update of release channel (41.0.1 etc.).

xabolcs commented:
@Cai0407, to be clear, this PR is for enable open-pushlog feature for all kind of release?

Cai0407 commented:
@xabolcs, I have tested open-pushlog feature under these situations and all work fine;

Firefox
1. "after current build" and "between last and current build"on 44 Nightly and 43 Developer Edition (mozilla-central, mozilla-aurora)
2. "after current build" on 42.0b5 and "between last and current build" from 42.0b4 to 42.0b5 (mozilla-beta)
3. "after current build" on 41.0.1 and "between last and current build" from 41.0 to 41.0.1 (mozilla-release)
4. "after current build" on 38.3.0 ESR and "between last and current build" from 38.2.1 ESR to 38.3.0 ESR (mozilla-esr38)

Thunderbird
1. "after current build" on 42.0b1 and "between last and current build" from 41.0b2 to 42.0b1 (comm-beta)
2. "after current build" on 38.3.0 and "between last and current build" from 38.2.0 to 38.3.0 (comm-esr38)

xabolcs commented:
In this case, it's good enough to remove the `isTrunk()` calls. No need for other touch.

Please remove the call from `menuPopup` function near [line 297][1], and revert all other changes (except the other `isTrunk()` [removal][2] from yesterday).

@whimboo, I'm unsure about the rest. `nightly.isTrunk()`, `nightlyApp.repository` would be unused if this lands. Should @Cai0407 also remove those?


[1]: /mozilla/nightlytt/blob/497220120678f574d317a4ac576c5c7fd4b46bb1/extension/chrome/content/nightly.js#L297
[2]: #issuecomment-147310693

whimboo commented on extension/chrome/content/browser.js:
```diff
@@ -4,7 +4,7 @@
 var nightlyApp = {
 
-repository: ['mozilla-central','mozilla-aurora'],
+repository: ['mozilla-central','mozilla-aurora','mozilla-beta','mozilla-release','mozilla-esr38','mozilla-esr31','mozilla-esr24','mozilla-esr17','mozilla-esr10'],
```
Everything below mozilla-esr38 is unsupported. No need to get those added. Also I wonder if we should better have a centralized config file to handle updates to those settings more easily.

whimboo commented on extension/chrome/content/messenger.js:
```diff
@@ -4,7 +4,7 @@
 var nightlyApp = {
 
-repository: ['comm-central','comm-aurora'],
+repository: ['comm-central','comm-aurora','comm-beta','comm-release','comm-esr38','comm-esr31','comm-esr24','comm-esr17','comm-esr10'],
```
Same here as above.

whimboo commented on extension/chrome/content/suite.js:
```diff
@@ -4,7 +4,7 @@
 var nightlyApp = {
 
-repository: ['comm-central','comm-aurora'],
+repository: ['comm-central','comm-aurora','comm-beta','comm-release','comm-esr38','comm-esr31','comm-esr24','comm-esr17','comm-esr10'],
```
And here too.

whimboo commented:
> @whimboo, I'm unsure about the rest. nightly.isTrunk(), nightlyApp.repository would be unused if this lands. Should @Cai0407 also remove those?

If we don't need those anymore let kill them for sure.

Cai0407 commented:
I have removed `nightly.isTrunk()` call from `menuPopup` function, and reverted all other changes for `isTrunk` in nightly.js  (except `isTrunk()` removal from `if` at line 134), and removed unsupported old branches from browser.js, messenger.js and suite.js.

Can we remove `isTrunk` function itself (delete lines 60-70 of nightly.js)?

xabolcs commented:
@Cai0407 sure we can remove `isTrunk`!
Please remove (now unused) `nightlyApp.repository` property too!

See @whimboo's [answer][1]!

Thanks!


[1]: #issuecomment-147705144

Cai0407 commented:
@xabolcs I have removed `isTrunk` function and `nightlyApp.repository` property.

xabolcs commented:
Thanks!
Changes looks good.

Somebody needs to test them! For example me! :)

In the mean time, @whimboo could you help providing a commit message for this PR?

xabolcs commented:
@whimboo!
Issue #51 should be referenced in the commit message.

xabolcs commented:
> Somebody needs to test them! For example me! :)

Tested with Mozilla/5.0 (X11; Linux x86_64; rv:38.0) Gecko/20100101 Firefox/38.0 ID:20150916094008 CSet: 7b1115ea78f7.

Looks good!

whimboo commented:
If testing showed that all is working fine I don't see any issues in getting this finally merged. @xabolcs mind doing it? I won't have the time today and will be away the next days.

whimboo commented:
> In the mean time, @whimboo could you help providing a commit message for this PR?

I would say lets take the summary of this PR? :)

xabolcs commented:
Thanks @whimboo! :)

Will land this today.

xabolcs commented:
Thanks @Cai0407!

Landed as 60e1ba747c321fe7012a2793d77b495da5a6ac4e.
